archive-data/archive.py

```diff
@@ -0,0 +1,204 @@
+"""This file is responsible for moving old data from
+the short-term storage to the long-term storage."""
+
+from functools import reduce
+import logging
+from os import environ as ENV
+
+from dotenv import load_dotenv
+from psycopg2 import connect, Error
+from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection
+import pandas as pd
+
+from archive_queries import (S_DELAYS, S_DELAYS_OVER_5_MIN, S_AVG_DELAY, S_TOTAL_ARRIVALS, S_TOTAL_CANCELLATIONS,
+                             O_DELAYS, O_DELAYS_OVER_5_MIN, O_AVG_DELAY, O_TOTAL_ARRIVALS, O_TOTAL_CANCELLATIONS,
+                             INSERT_STATION_PERFORMANCE, INSERT_OPERATOR_PERFORMANCE,
+                             DELETE_OLD_ARRIVAL_DATA, DELETE_OLD_CANCELLATION_DATA)
+
+
+class DataFetchError(Exception):
+    """Exception raised when there is an error fetching data."""
+    pass
+
+
+def setup_logging() -> None:
+    """Sets up the logging configuration."""
+
+    logging.basicConfig(level=logging.INFO,
+                        format='%(asctime)s:%(levelname)s:%(message)s')
+
+
+def get_db_connection(config: dict[str, str]) -> connection:
+    """Returns a connection to a database."""
+
+    return connect(
+        host=config["DB_HOST"],
+        user=config["DB_USER"],
+        password=config["DB_PASS"],
+        dbname=config["DB_NAME"],
+        port=config["DB_PORT"],
+        cursor_factory=RealDictCursor
+    )
+
+
+def close_connection(conn: connection) -> None:
+    """Closes the connection to the database."""
+
+    conn.close()
+
+
+def fetch_old_data(conn: connection, query: str) -> pd.DataFrame:
+    """Fetches historical data based on the query.
+    Returns the fetched data as a DataFrame."""
+
+    try:
+
+        with conn.cursor() as cur:
+            cur.execute(query)
+            rows = cur.fetchall()
+        return pd.DataFrame(rows)
+
+    except Error as err:
```

Comment:
Nitpick: Maybe you want to specify what type of error you're expecting?

Reply:
Error relates to errors from psycopg2 (`from psycopg2 import connect, Error`)